src/Discord.Net.Core/Entities/SlashCommands/InteractionToken.cs

```diff
@@ -0,0 +1,34 @@
+using System;
+
+namespace Discord
+{
+    /// <inheritdoc/>
+    public class InteractionToken : IDiscordInteractionToken
```

Comment:
Current Token validation function seems unreliable since it uses the token arrival time. This class might be redundant but i think we should still use the Snowflake creation time for validating the token